File: tests/fixtures/js_ts/app/api/controllers/userController.ts
import { UserService } from '../../services/userService';

export const getUserProfile = (
    userService: UserService,
    userId: number,
): { status: number; body: Record<string, unknown> } => {
    const profile = userService.loadProfile(userId);
    if (!profile) {
        return { status: 404, body: { error: 'not_found' } };
    }
    return { status: 200, body: profile };
};

export const getUserProfileAsync = async (
    userService: UserService,
    userId: number,
    token: string,
): Promise<{ status: number; body: Record<string, unknown> }> => {
    const result = await userService.loadProfileAsync(userId, token);
    if (!result.ok) {
        return { status: 401, body: { error: 'unauthorized' } };
    }
    return { status: 200, body: result.data };
};

File: tests/fixtures/py/app/api/controllers/user_controller.py
"""Controller layer for Python fixture app."""

from __future__ import annotations

from app.api.serializers.profile_serializer import to_http_response
from app.services.user_service import UserService


def get_user_profile(user_service: UserService, user_id: int) -> dict[str, object]:
    """Return HTTP response for profile endpoint."""
    profile = user_service.load_profile(user_id)
    return to_http_response(profile)


async def get_user_profile_async(
    user_service: UserService, user_id: int
) -> dict[str, object]:
    """Return async HTTP response for profile endpoint."""
    profile = await user_service.load_profile_async(user_id)
    return to_http_response(profile)
